openamp/0007-openamp-change-can_recursive-to-wait_tx_buffer.patch

```diff
@@ -0,0 +1,153 @@
+From 8b5c3bf79c3165ea4068fc52a46f3cad7c6c5ead Mon Sep 17 00:00:00 2001
+From: ligd <liguiding1@xiaomi.com>
+Date: Mon, 29 Nov 2021 21:45:30 +0800
+Subject: [PATCH 1/3] openamp: change can_recursive to wait_tx_buffer
+
+Signed-off-by: ligd <liguiding1@xiaomi.com>
+---
+ lib/include/openamp/remoteproc.h        |  8 ++++----
+ lib/include/openamp/remoteproc_virtio.h | 11 -----------
+ lib/include/openamp/rpmsg_virtio.h      |  6 ++++++
+ lib/include/openamp/virtio.h            |  1 +
+ lib/remoteproc/remoteproc_virtio.c      | 23 ++++++++++++-----------
+ lib/rpmsg/rpmsg_virtio.c                |  4 ++--
+ 6 files changed, 25 insertions(+), 28 deletions(-)
+
+diff --git a/lib/include/openamp/remoteproc.h openamp/lib/include/openamp/remoteproc.h
+index 222079e..328ec3f 100644
+--- a/lib/include/openamp/remoteproc.h
++++ openamp/lib/include/openamp/remoteproc.h
+@@ -450,15 +450,15 @@ struct remoteproc_ops {
+ 					  void *va, size_t size,
+ 					  struct remoteproc_mem *buf);
+ 	/**
+-	 * can_recursive
++	 * wait_tx_buffer
+ 	 *
+-	 * Check current vdev can do recursive rx_callabck or not
++	 * Wait tx buffer available
+ 	 *
+ 	 * @rproc - pointer to remoteproc instance
+ 	 *
+-	 * return true if can do recursive rx_callabck.
++	 * return none
+ 	 */
+-	int (*can_recursive)(struct remoteproc *rproc);
```

Comment:
`can_recursive` is introduced by `0006-openamp-fix-scenario-case.patch`. Could you please rework it instead of creating patch for the patch?

Reply:
Good, done